samples/extensions/sparse_image/sparse_image.h

```diff
@@ -0,0 +1,245 @@
+/* Copyright (c) 2023, Mobica Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "api_vulkan_sample.h"
+
+size_t const ON_SCREEN_HORIZONTAL_BLOCKS = 100;
+size_t const ON_SCREEN_VERTICAL_BLOCKS   = 80;
+double const FOV_DEGREES                 = 60.0;
+double const MIP_LEVEL_MARGIN            = 0.3;
+
+class sparse_image : public ApiVulkanSample
+{
+  public:
+
+	struct MVP
+	{
+		alignas(16) glm::mat4 model;
+		alignas(16) glm::mat4 view;
+		alignas(16) glm::mat4 proj;
+	};
+
+	struct SimpleBuffer
```

Comment:
This structure is not needed and not using it correctly can cause performance problems. Use vkb::core::Buffer, which uses VMA.